Dataset Documentation
Dataset Name: Pakistan Job Market Intelligence Dataset
Source URL: https://www.kaggle.com/datasets/aamir28/pakistan-job-market-intelligence-dataset
Number of Rows: 10500
Number of Columns: 13
Time Period: May 2024 to May 2025
Collected By: Kaggle user aamir28, calibrated against real data from Rozee.pk, LinkedIn Pakistan, Indeed PK and Mustakbil.
Column Descriptions
Job Title: The title of the job being advertised
Company: Name of the company posting the job
City: City where the job is located
Sector: Industry sector of the job
Salary Range PKR: Monthly salary range offered in Pakistani Rupees
Required Skills: Skills needed for the job
Experience Required: Years of experience required
Job Type: Whether job is Onsite, Hybrid or Remote
Education Level: Minimum education qualification required
Gender Preference: Whether employer prefers Male, Female or No Preference
Number of Vacancies: How many positions are available
Posted Date: Date when the job was posted
Application Deadline: Last date to apply for the job

In [2]:
import pandas as pd
df = pd.read_csv("pakistan_job_market_dataset.csv")

In [4]:
df.shape

(10500, 13)

In [5]:
df.dtypes

Job Title               object
Company                 object
City                    object
Sector                  object
Salary Range (PKR)      object
Required Skills         object
Experience Required     object
Job Type                object
Education Level         object
Gender Preference       object
Number of Vacancies      int64
Posted Date             object
Application Deadline    object
dtype: object

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10500 entries, 0 to 10499
Data columns (total 13 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Job Title             10500 non-null  object
 1   Company               10500 non-null  object
 2   City                  10500 non-null  object
 3   Sector                10500 non-null  object
 4   Salary Range (PKR)    10185 non-null  object
 5   Required Skills       10500 non-null  object
 6   Experience Required   10395 non-null  object
 7   Job Type              10500 non-null  object
 8   Education Level       10290 non-null  object
 9   Gender Preference     10343 non-null  object
 10  Number of Vacancies   10500 non-null  int64 
 11  Posted Date           10500 non-null  object
 12  Application Deadline  10500 non-null  object
dtypes: int64(1), object(12)
memory usage: 1.0+ MB


In [10]:
df.describe()

,Number of Vacancies
count,10500.000000
mean,3.091810
std,3.659364
min,1.000000
25%,1.000000
50%,2.000000
75%,3.000000
max,20.000000


In [12]:
df.columns

Index(['Job Title', 'Company', 'City', 'Sector', 'Salary Range (PKR)',
       'Required Skills', 'Experience Required', 'Job Type', 'Education Level',
       'Gender Preference', 'Number of Vacancies', 'Posted Date',
       'Application Deadline'],
      dtype='object')

In [14]:
cols = ['City', 'Sector', 'Experience Required', 'Job Type',
         'Education Level', 'Gender Preference']

print(df['City'].value_counts())

City
Karachi       2181
Lahore        1991
Islamabad     1479
Rawalpindi     759
Faisalabad     629
Peshawar       610
Multan         540
Sialkot        427
Quetta         420
Hyderabad      339
Gujranwala     283
Abbottabad     215
Bahawalpur     205
Sukkur         127
Karachì         46
IslamaBad       41
lahore          38
karachi         37
islamabad       36
LAHORE          33
Lahorë          33
KARACHI         31
Name: count, dtype: int64


In [15]:
print(df['Sector'].value_counts())

Sector
IT & Technology               1889
Banking & Finance             1130
Education                     1070
Healthcare                     939
E-commerce & Retail            811
Telecom                        713
Government & Public            700
Manufacturing                  700
FMCG & Consumer Goods          616
Logistics & Supply Chain       546
Construction & Real Estate     529
NGO & Development              429
Media & Marketing              428
Name: count, dtype: int64


In [16]:
print(df['Experience Required'].value_counts())

Experience Required
1-3 years    3709
3-5 years    2877
Fresh        2094
5+ years     1715
Name: count, dtype: int64


In [17]:
print(df['Job Type'].value_counts())

Job Type
Onsite    6778
Hybrid    1957
Remote    1765
Name: count, dtype: int64


In [18]:
print(df['Education Level'].value_counts())

Education Level
Bachelor's      5168
Master's        2781
MBA              743
Intermediate     599
MBBS             348
MPhil            292
PhD              212
Matric           147
Name: count, dtype: int64


In [19]:
df.isnull().sum()

Job Title                 0
Company                   0
City                      0
Sector                    0
Salary Range (PKR)      315
Required Skills           0
Experience Required     105
Job Type                  0
Education Level         210
Gender Preference       157
Number of Vacancies       0
Posted Date               0
Application Deadline      0
dtype: int64

In [20]:
#percentage missing values
print((df.isnull().sum()/len(df)) *100)

Job Title               0.000000
Company                 0.000000
City                    0.000000
Sector                  0.000000
Salary Range (PKR)      3.000000
Required Skills         0.000000
Experience Required     1.000000
Job Type                0.000000
Education Level         2.000000
Gender Preference       1.495238
Number of Vacancies     0.000000
Posted Date             0.000000
Application Deadline    0.000000
dtype: float64


In [21]:
#IQR method
Q1 = df["Number of Vacancies"].quantile(0.25)
Q3  = df["Number of Vacancies"].quantile(0.75)
IQR = Q3-Q1

lower = Q1-1.5 * IQR
upper = Q3 +1.5 *IQR

outliers = df[df["Number of Vacancies"] > upper]
print(f"Total Outliers : {len(outliers)}")
print(f"Upper bound : {upper}")

Total Outliers : 928
Upper bound : 6.0


In [22]:
# Required Skills is hard to use directly - needs splitting and exploding for proper analysis
# Application Deadline is redundant - Posted Date is sufficient for time-based analysis

8 specific questions about the dataset purely through code and print statements

In [24]:
#Q1."Which city has the most job postings?"
print(df["City"].value_counts().idxmax())

Karachi


In [25]:
#Q2 "Which sector has the most job postings?"
print(df["Sector"].value_counts().idxmax())

IT & Technology


In [26]:
#Q3 "What percentage of jobs are remote?"

remote = df[df["Job Type"]=='Remote']
print(len(remote)/len(df) * 100)

16.80952380952381


In [27]:
#Q4.  "Which sector has the highest average number of vacancies?"
print(df.groupby('Sector')['Number of Vacancies'].mean().idxmax())

Construction & Real Estate


In [28]:
#Q5 "What is the most required education level?"
print(df['Education Level'].value_counts().idxmax())


Bachelor's


In [29]:
#Q6 "Which sector pays the highest salary?"

df['Salary Avg'] = df['Salary Range (PKR)'].str.replace(',', '').str.split(' - ').apply(lambda x: (int(x[0]) + int(x[1])) / 2 if isinstance(x, list) and len(x) == 2 else None)
print(df.groupby('Sector')['Salary Avg'].mean().idxmax())

IT & Technology


In [30]:
#Q7"What percentage of jobs prefer male candidates?"
male = df[df["Gender Preference"]=="Male"]
print(len(male)/len(df) * 100)

11.780952380952382


In [31]:
#Q8 "Which city has the highest average salary?"
print(df.groupby('City')['Salary Avg'].mean().idxmax())

islamabad


Data Quality Report
This dataset contains 10500 rows and 13 columns covering Pakistan job market data from May 2024 to May 2025 collected from major Pakistani hiring platforms.
Missing Values
Salary Range column has 3 percent missing values. Education Level has 2 percent missing values. Gender Preference has 1.5 percent missing values. Experience Required has 1 percent missing values. All missing values are under 5 percent so no column needs to be dropped. In Week 2 these will be filled using mode imputation.
Data Entry Errors
City column has serious spelling inconsistencies. Karachi is written as Karachì, karachi and KARACHI. Lahore appears as lahore, LAHORE and Lahorë. Islamabad appears as IslamaBad and islamabad. These are all the same cities but pandas treats them as different values which causes wrong counts. In Week 2 all city names will be standardized to proper title case.
Outliers
Number of Vacancies is the only numeric column. Using the IQR method the upper bound was calculated as 6. Any job posting with more than 6 vacancies is flagged as an outlier. Total 928 outlier rows were found. These are not necessarily errors since a company can genuinely need many hires at once. They will be investigated further in Week 2.
Redundant and Problematic Columns
Application Deadline column is redundant because Posted Date already provides time information. It will be dropped in Week 2. Required Skills column contains comma separated skill lists in a single cell which makes direct analysis impossible. In Week 2 this column will be split and exploded into individual rows for proper skill frequency analysis.
